doc/examples/affine_registration_3d.py

```diff
@@ -0,0 +1,236 @@
+"""
+==========================================
+Affine Registration in 3D
+==========================================
+This example explains how to compute an affine transformation to register two 3D
+volumes by maximization of their Mutual Information [Mattes03]_. The optimization
+strategy is similar to that implemented in ANTS [Avants11]_.
+"""
+
+import numpy as np
+import nibabel as nib
+import os.path
+from dipy.viz import regtools
+from dipy.data import fetch_stanford_hardi, read_stanford_hardi
+from dipy.data.fetcher import fetch_syn_data, read_syn_data
+from dipy.segment.mask import median_otsu
+from dipy.align.imaffine import (align_centers_of_mass,
+                                 transform_image,
+                                 MattesMIMetric,
+                                 AffineRegistration)
+from dipy.align.transforms import regtransforms
+
+"""
+Let's fetch two b0 volumes, the static image will be the b0 from the Stanford
+HARDI dataset
+"""
+
+fetch_stanford_hardi()
+nib_stanford, gtab_stanford = read_stanford_hardi()
+static = np.squeeze(nib_stanford.get_data())[..., 0]
+static_grid2space = nib_stanford.get_affine()
+
+"""
+Now the moving image
+"""
+
+fetch_syn_data()
+nib_syn_t1, nib_syn_b0 = read_syn_data()
+moving = np.array(nib_syn_b0.get_data())
+moving_grid2space = nib_syn_b0.get_affine()
+
+"""
+We can obtain a very rough (and fast) registration by just aligning the centers of mass
+of the two images
+"""
+
+c_of_mass = align_centers_of_mass(static, static_grid2space, moving, moving_grid2space)
+
+"""
+We can now warp the moving image and draw it on top of the static image, registration
+is not likely to be good, but at least they will occupy roughly the same space
+"""
+
+warped = transform_image(static, static_grid2space, moving, moving_grid2space, c_of_mass)
+regtools.overlay_slices(static, warped, None, 0, "Static", "Warped", "warped_com_0.png")
+regtools.overlay_slices(static, warped, None, 1, "Static", "Warped", "warped_com_1.png")
+regtools.overlay_slices(static, warped, None, 2, "Static", "Warped", "warped_com_2.png")
```

Comment:
What I don't like in the overlay slices is that you are not showing the moving image.  It is very difficult to see the difference just by looking to red/green overlay. This is a problem that we have in the diffeomorphic examples too. The information communicates much better when you show the moving image too and if you show the transformation on the grid that is even better. We have to make sure that we clearly show the benefits of using your tools.